azure-kusto-data/azure/kusto/data/data_format.py

```diff
@@ -2,22 +2,37 @@
 # Licensed under the MIT License
 from enum import Enum
 
+class IngestionMappingType(Enum):
+    CSV = "Csv"
+    JSON = "Json"
+    AVRO = "Avro"
+    PARQUET = "Parquet"
+    ORC = "Orc"
+    W3CLOGFILE = "W3CLogFile"
+    UNKNOWN = "Unknown"
+
 
 class DataFormat(Enum):
     """All data formats supported by Kusto."""
 
-    CSV = "csv"
-    TSV = "tsv"
-    SCSV = "scsv"
-    SOHSV = "sohsv"
-    PSV = "psv"
-    TXT = "txt"
-    JSON = "json"
-    SINGLEJSON = "singlejson"
-    AVRO = "avro"
-    PARQUET = "parquet"
-    MULTIJSON = "multijson"
-    ORC = "orc"
-    TSVE = "tsve"
-    RAW = "raw"
-    W3CLOGFILE = "w3clogfile"
+    CSV = ("csv", IngestionMappingType.CSV, False, True)
+    TSV = ("tsv", IngestionMappingType.CSV, False, True)
+    SCSV = ("scsv", IngestionMappingType.CSV, False, True)
+    SOHSV = ("sohsv", IngestionMappingType.CSV, False, True)
+    PSV = ("psv", IngestionMappingType.CSV, False, True)
+    TXT = ("txt", IngestionMappingType.UNKNOWN, False, True)
+    TSVE = ("tsve", IngestionMappingType.CSV, False, True)
+    JSON = ("json", IngestionMappingType.JSON, True, True)
+    SINGLEJSON = ("singlejson", IngestionMappingType.JSON, True, True)
+    MULTIJSON = ("multijson", IngestionMappingType.JSON, True, True)
+    AVRO = ("avro", IngestionMappingType.AVRO, True, True)
+    PARQUET = ("parquet", IngestionMappingType.PARQUET, False, False)
+    ORC = ("orc", IngestionMappingType.ORC, False, False)
+    RAW = ("raw", IngestionMappingType.UNKNOWN, False, True)
+    W3CLOGFILE = ("w3clogfile", IngestionMappingType.W3CLOGFILE, False, True)
+
+    def __init__(self, value_: str, ingestion_mapping_type: IngestionMappingType, mapping_required: bool, compressible: bool):
+        self.value_ = value_ # Formatted how Kusto Service expects it
+        self.ingestion_mapping_type = ingestion_mapping_type
+        self.mapping_required = mapping_required
+        self.compressible = compressible # Binary formats should not be compressed
```

Comment:
I didn't realize (until the merge conflicts now) that Asaf recently made some changes to is_mapping_required and is_binary/is_compressible. I don't think his PR affects any of my changes here, except for a decision regarding whether we want is_binary or is_compressible. I'm open to either, but prefer the former if we have any reason to care whether a format is binary other than to determine if it's compressible, and otherwise prefer the latter.

I will discuss this with relevant people.

Reply:
is_binary was only used for checking if to compress.  
I can't really think of another reason to want this information, so I'm fine with just is_compressible